lib/open_food_network/scope_variant_to_hub.rb

```diff
@@ -28,7 +28,8 @@ def in_stock?
         on_demand || (count_on_hand > 0)
       end
 
-      def count_on_hand
```

Comment:
no, it defaults to variant.count_on_hand that is implemented in variant_stock that calls variant.total_on_hand.

I am answering you longer comment here:
first, your explanation is correct!
Did you see my previous versions of this PR? your suggestion is where I started, really! Having the code in stock_location_decorator.

I didn't try to change the way overrides are handled, I wouldnt go that far.
Also, the attribute you propose variant.override imo is a very bad pattern, linking from parent to child would make the model even worse.
I dont think any of this code should be in stock_location (I think we have enough complicated decorated spree classes) that's why I moved it into variant_stock (with this PR I think variant stock has become a OK class actually, I think it makes sense as an unit).
I am quite happy with my solution, mainly because:
- it makes use of the inheritance implied in the name Variant Override
- it keeps stock location very clean (one line method)
- the variant.move really makes sense in variant_stock specially as we have the overwrite_stock_level below

Back to your main concern: imo having 2 methods move is really ok here because we are using the inheritance mechanism that "override" implies.

Reply:
There's one more main advantage I see in putting things in variant and variant_override instead of stock location: it's encapsulation. Stock_location shouldn't know about overrides, that complexity should be kept in it's own place, so variant knows about its things (single stock item per variant for example) and overrides about theirs (when to use their stuff or call super).

Reply:
I like that we have a bit of timezone overlap to get a quick response. :-)

It sounds reasonable to keep the current concept of variant overrides, not introducing new approaches here. I still think that we are building on a bad design and now introduce a new method within that bad design which is not necessary. But it's okay, it fits into the pattern. I just needed half an hour to understand what is actually happening when we try to change the stock level. We will get rid of this though. The Network feature has to come soon to end this.

![The end is near](https://media.giphy.com/media/3orieVLyNQr2TvLDs4/giphy.gif)

I still have one question: The original Spree code creates a stock item if none is present. In your version we raise an error instead. Is that just legacy behaviour, because we didn't know what to do when we created VariantStock? Does it make sense or should we adopt Spree's approach of creating missing stock items?

Reply:
> Stock_location shouldn't know about overrides, that complexity should be kept in it's own place

I totally agree that StockLocation shouldn't know about it. But I don't think `Spree::Variant` should know about it either, it's a Spree model. I was thinking about an OFN `StockUpdater` that knows about the relationship between variants, enterprises and stock levels. In the network feature we may have modelled it better so that we actually don't need our own stock updater any more. So let's forget about this. ;-)

Reply:
I agree. You already have a comment in VariantStock to explain that it's the original code from StockLocation#move. It would be good to clarify there that it's not exactly the same and why.

```diff
- # Creates a stock movement (this is the original spree stock_location.move)
+ # Creates a stock movement. This is the original Spree::StockLocation#move.
+ # The only difference is that we are raising an error if the stock item is missing,
+ # because we should always have exactly one stock item per variant (unlike Spree).
```